samples/net/sockets/http_server/src/server.c

```diff
@@ -0,0 +1,721 @@
+/*
+ * Copyright (c) 2023, Emna Rekik
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/net/http/parser.h>
+#include <zephyr/net/http/server.h>
+#include <zephyr/net/socket.h>
+#include <zephyr/posix/arpa/inet.h>
+#include <zephyr/posix/fcntl.h>
+#include <zephyr/posix/netinet/in.h>
+#include <zephyr/posix/poll.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/posix/sys/eventfd.h>
```

Comment:
Just remember to add `CONFIG_POSIX_API=y` to your `prj.conf`.

Also, please separate the headers into the appropriate canonical blocks.

```suggestion
#include <sys/socket.h>
#include <arpa/inet.h>
#include <fcntl.h>
#include <netinet/in.h>
#include <poll.h>
#include <zephyr/sys/atomic.h>
#include <sys/eventfd.h>
```